libraries/chain/services/reward_balancer.cpp

```diff
@@ -0,0 +1,108 @@
+#include <scorum/chain/services/reward_balancer.hpp>
+#include <scorum/chain/database/database.hpp>
+
+#include <scorum/chain/schema/reward_balancer_object.hpp>
+#include <scorum/chain/schema/scorum_objects.hpp>
+
+namespace scorum {
+namespace chain {
+
+dbs_reward::dbs_reward(database& db)
+    : _base_type(db)
+{
+}
+
+const reward_balancer_object& dbs_reward::create_pool(const asset& initial_supply)
+{
+    // clang-format off
+    FC_ASSERT(db_impl().find<reward_balancer_object>() == nullptr, "recreation of reward_balancer_object is not allowed");
+    FC_ASSERT(initial_supply.symbol() == SCORUM_SYMBOL, "initial supply for reward_pool must in ${1}", ("1", asset(0, SCORUM_SYMBOL).symbol_name()));
+
+    return db_impl().create<reward_balancer_object>([&](reward_balancer_object& rp) {
+        rp.balance = initial_supply;
+    });
+    // clang-format on
+}
+
+bool dbs_reward::is_pool_exists() const
+{
+    return nullptr != db_impl().find<reward_balancer_object>();
+}
+
+const reward_balancer_object& dbs_reward::get_pool() const
+{
+    return db_impl().get<reward_balancer_object>();
+}
+
+void dbs_reward::update(const modifier_type& modifier)
+{
+    db_impl().modify(get_pool(), [&](reward_balancer_object& o) { modifier(o); });
+}
+
+const asset& dbs_reward::increase_pool_ballance(const asset& delta)
+{
+    const auto& pool = get_pool();
+
+    db_impl().modify(pool, [&](reward_balancer_object& pool) {
+        switch (delta.symbol())
+        {
+        case SCORUM_SYMBOL:
+            FC_ASSERT(delta >= asset(0, SCORUM_SYMBOL));
+            pool.balance += delta;
+            break;
+        default:
+            FC_ASSERT(false, "invalid symbol");
+        }
+    });
+
+    return pool.balance;
+}
+
+// clang-format off
+const asset dbs_reward::take_block_reward()
```

Comment:
This is not the method for dbs_service. This is method for task class. If you have a time fix this.
It is not critical but TODO